src/pages/RiddleSolvingPage.tsx

```diff
@@ -0,0 +1,49 @@
+import type { FC } from 'react';
+import { useParams } from 'react-router-dom';
+import { Card, CardContent, CardMedia, Stack, Typography } from '@mui/material';
+
+import {
+	MockDisplayQuestions,
+	MockRiddleDisplayDetails
+} from '../../mock-data/MockData';
+import { QuestionSolvingAccordion } from '../components/forms/solvingForm/QuestionSolvingAccordion';
+import type { RiddleDisplayDetail } from '../utils/Types';
+
+export const RiddleSolvingPage: FC = () => {
+	//TODO: replace with link id
+	const { id } = useParams();
+	// @ts-ignore TODO: fetch data about the riddle
+	const riddleData: RiddleDisplayDetail = MockRiddleDisplayDetails[id - 1];
+	return (
+		<Stack gap={2}>
+			<Typography variant="h4" fontWeight="bold">
+				{riddleData.name}
+			</Typography>
+			{/* TODO: replace with data fetch */}
+			{MockDisplayQuestions.map(question => (
+				<QuestionSolvingAccordion {...question} key={question.id} />
+			))}
+			<Card sx={{ backgroundColor: 'background.default' }}>
```

Comment:
Displaying the correct solution differs from Figma since we have an optional text & image to be displayed when the whole riddle is solved, while the correct answers correspond to particular questions (not the whole riddle).

Reply:
Chybí tu podmínka zobrazení jen tehdy, když je celá hádanka vyřešená. Přidám po mergnutí ve věci integrující s db.

Reply:
Bylo to myšlené tak, že po zodpovězení všech otázek se zobrazí dialog s šifrou. Ale minimálně pro teď to nechme takto.

Reply:
No to je ale přesně to, co nemáme, ne?
Otázky nemusí nutně odkrýt jednu odpověď, to záleží, co tam vyplní autor, jestli vůbec něco.

Reply:
Asi nerozumím. :D
Myslel jsem, že je to `solvedText` (popř. `solvedImage`).